src/Products/PluginIndexes/DateIndex/tests.py

```diff
@@ -184,9 +184,9 @@ def _convert(self, dt, precision=1):
         from Products.PluginIndexes.DateIndex.DateIndex import Local
         if isinstance(dt, (float, int)):
             yr, mo, dy, hr, mn = gmtime(dt)[:5]
-        elif type(dt) is date:
+        elif isinstance(dt, date):
```

Comment:
The tests are broken because of this change: a `datetime` object is also a `date` by inheritance. So `type(dt) is date` is different than `isinstance(dt, date)` when talking about a `datetime`.

Anyway IMO tests could be easy fixed changing `if` order, as:

```
         from Products.PluginIndexes.DateIndex.DateIndex import Local
         if isinstance(dt, (float, int)):
             yr, mo, dy, hr, mn = gmtime(dt)[:5]
-        elif isinstance(dt, date):
-            yr, mo, dy, hr, mn = dt.timetuple()[:5]
         elif isinstance(dt, datetime):
             if dt.tzinfo is None:  # default behavior of index
                 dt = dt.replace(tzinfo=Local)
             yr, mo, dy, hr, mn = dt.utctimetuple()[:5]
+        elif isinstance(dt, date):
+            yr, mo, dy, hr, mn = dt.timetuple()[:5]
         else:
             yr, mo, dy, hr, mn = dt.toZone('UTC').parts()[:5]
         value = (((yr * 12 + mo) * 31 + dy) * 24 + hr) * 60 + mn

```

Reply:
In the end here https://github.com/zopefoundation/Products.ZCatalog/blob/master/src/Products/PluginIndexes/DateIndex/DateIndex.py#L125 the order is as I suggested.

Reply:
I applied the fix you suggested in https://github.com/zopefoundation/Products.ZCatalog/pull/143/commits/56d17f7363b7c6c7903af24936c24bc7661deb36